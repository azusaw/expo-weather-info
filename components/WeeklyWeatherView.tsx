import React from "react";
import { Dimensions, FlatList, StyleSheet } from "react-native";
import { Text, View } from "@/components/Themed";
import DailyWeatherCard from "@/components/DailyWeatherCard";
import { DailyWeather } from "@/types";
import Colors from "@/constants/Colors";

const { width } = Dimensions.get("window");

const WeeklyWeatherView = ({ data }: { data: DailyWeather[] }) => (
  <View style={styles.weeklyWeather}>
    <View style={styles.title}>
      <Text size={20} weight={300} color={Colors.text.light}>
        {"Weekly forecast"}
      </Text>
      <View style={styles.arrowContainer}>
        <View style={[styles.arrow, styles.arrowHorizontal]} />
        <View style={[styles.arrow, styles.arrowDiagonal]} />
      </View>
    </View>
    <FlatList
      horizontal
      data={data}
      renderItem={({ item }) => <DailyWeatherCard {...item} />}
      keyExtractor={(item) => item.time}
      contentContainerStyle={{ paddingHorizontal: 10 }}
    />
  </View>
);

const styles = StyleSheet.create({
  weeklyWeather: {
    maxWidth: width,
  },
  title: {
    marginTop: 20,
    marginBottom: 10,
    paddingHorizontal: 20,
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
  },
  arrowContainer: {
    position: "relative",
    backgroundColor: "transparent",
    width: 60,
    height: 50,
    right: 0,
  },
  arrow: {
    backgroundColor: Colors.text.light,
    position: "absolute",
    height: 1,
    borderRadius: 5,
  },
  arrowHorizontal: {
    width: 58,
    bottom: "40%",
  },
  arrowDiagonal: {
    width: 15,
    bottom: "50%",
    right: 0,
    transform: [{ rotate: "45deg" }],
  },
});

export default WeeklyWeatherView;
