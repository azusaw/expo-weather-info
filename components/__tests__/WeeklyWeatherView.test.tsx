import { render } from "@testing-library/react-native";
import WeeklyWeatherView from "@/components/WeeklyWeatherView";
const mockProps = {
  data: [
    {
      time: "2025-04-16T23:45",
      weatherCode: 2,
      temperatureMin: 12.5,
      temperatureMax: 19.6,
    },
    {
      time: "2025-04-17T23:45",
      weatherCode: 7,
      temperatureMin: 14.5,
      temperatureMax: 16.6,
    },
    {
      time: "2025-04-18T23:45",
      weatherCode: 23,
      temperatureMin: 10.5,
      temperatureMax: 16.8,
    },
  ],
};

describe("WeeklyWeatherView", () => {
  test("it renders WeeklyWeatherView successfully", () => {
    const { toJSON } = render(<WeeklyWeatherView {...mockProps} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test("it renders heading", () => {
    const { getByText } = render(<WeeklyWeatherView {...mockProps} />);
    expect(getByText("Weekly forecast")).toBeTruthy();
  });

  test("it renders correct number of DailyWeather Card", () => {
    const { getByText } = render(<WeeklyWeatherView {...mockProps} />);
    ["16 Wed", "17 Thu", "18 Fri"].forEach((data) =>
      expect(getByText(data)).toBeTruthy(),
    );
  });
});
