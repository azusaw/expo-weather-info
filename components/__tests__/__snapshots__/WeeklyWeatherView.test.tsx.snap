// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeeklyWeatherView it renders WeeklyWeatherView successfully 1`] = `
<View
  style={
    [
      {
        "height": 280,
        "maxWidth": 0,
      },
    ]
  }
>
  <View
    style={
      [
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 10,
            "marginTop": 20,
            "paddingHorizontal": 20,
          },
          false,
        ],
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#eee",
            "fontFamily": "Montserrat_300Light",
            "fontSize": 20,
          },
          undefined,
        ]
      }
    >
      Weekly forecast
    </Text>
    <View
      style={
        [
          {
            "backgroundColor": "transparent",
            "height": 50,
            "position": "relative",
            "right": 0,
            "width": 60,
          },
        ]
      }
    >
      <View
        style={
          [
            [
              {
                "backgroundColor": "#eee",
                "borderRadius": 5,
                "height": 1,
                "position": "absolute",
              },
              {
                "bottom": "40%",
                "width": 58,
              },
            ],
          ]
        }
      />
      <View
        style={
          [
            [
              {
                "backgroundColor": "#eee",
                "borderRadius": 5,
                "height": 1,
                "position": "absolute",
              },
              {
                "bottom": "50%",
                "right": 0,
                "transform": [
                  {
                    "rotate": "45deg",
                  },
                ],
                "width": 15,
              },
            ],
          ]
        }
      />
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      {
        "height": "100%",
        "paddingHorizontal": 10,
      }
    }
    data={
      [
        {
          "temperatureMax": 19.6,
          "temperatureMin": 12.5,
          "time": "2025-04-16T23:45",
          "weatherCode": 2,
        },
        {
          "temperatureMax": 16.6,
          "temperatureMin": 14.5,
          "time": "2025-04-17T23:45",
          "weatherCode": 7,
        },
        {
          "temperatureMax": 16.8,
          "temperatureMin": 10.5,
          "time": "2025-04-18T23:45",
          "weatherCode": 23,
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            [
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff30",
                  "borderRadius": 10,
                  "elevation": 10,
                  "margin": 10,
                  "padding": 20,
                  "shadowColor": "#dddddd30",
                  "shadowOpacity": 0.8,
                  "shadowRadius": 10,
                },
                false,
              ],
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#eee",
                  "fontFamily": "Montserrat_400Regular",
                  "fontSize": 20,
                },
                undefined,
              ]
            }
          >
            16 Wed
          </Text>
          <View
            style={
              [
                {
                  "alignItems": "center",
                },
              ]
            }
          >
            <ViewManagerAdapter_ExpoImage
              containerViewRef={"[React.ref]"}
              contentFit="cover"
              contentPosition={
                {
                  "left": "50%",
                  "top": "50%",
                }
              }
              height={70}
              nativeViewRef={"[React.ref]"}
              onError={[Function]}
              onLoad={[Function]}
              onLoadStart={[Function]}
              onProgress={[Function]}
              placeholder={[]}
              source={
                [
                  {
                    "uri": "http://openweathermap.org/img/wn/02d@2x.png",
                  },
                ]
              }
              style={
                {
                  "height": 70,
                  "width": 80,
                }
              }
              transition={
                {
                  "duration": 1000,
                }
              }
              width={80}
            />
            <Text
              style={
                [
                  {
                    "color": "#eee",
                    "fontFamily": "Montserrat_300Light",
                    "fontSize": 14,
                  },
                  undefined,
                ]
              }
            >
              Partly Cloudy
            </Text>
          </View>
          <Text
            style={
              [
                {
                  "color": "#eee",
                  "fontFamily": "Montserrat_500Medium",
                  "fontSize": 16,
                },
                {
                  "marginTop": 10,
                },
              ]
            }
          >
            12.5
            <Text
              style={
                [
                  {
                    "color": "#eee",
                    "fontFamily": "Montserrat_200ExtraLight",
                    "fontSize": 14,
                  },
                  {
                    "marginHorizontal": 10,
                  },
                ]
              }
            >
              /
            </Text>
            19.6
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            [
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff30",
                  "borderRadius": 10,
                  "elevation": 10,
                  "margin": 10,
                  "padding": 20,
                  "shadowColor": "#dddddd30",
                  "shadowOpacity": 0.8,
                  "shadowRadius": 10,
                },
                false,
              ],
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#eee",
                  "fontFamily": "Montserrat_400Regular",
                  "fontSize": 20,
                },
                undefined,
              ]
            }
          >
            17 Thu
          </Text>
          <Text
            style={
              [
                {
                  "color": "#eee",
                  "fontFamily": "Montserrat_500Medium",
                  "fontSize": 16,
                },
                {
                  "marginTop": 10,
                },
              ]
            }
          >
            14.5
            <Text
              style={
                [
                  {
                    "color": "#eee",
                    "fontFamily": "Montserrat_200ExtraLight",
                    "fontSize": 14,
                  },
                  {
                    "marginHorizontal": 10,
                  },
                ]
              }
            >
              /
            </Text>
            16.6
          </Text>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={
          [
            {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          style={
            [
              [
                {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff30",
                  "borderRadius": 10,
                  "elevation": 10,
                  "margin": 10,
                  "padding": 20,
                  "shadowColor": "#dddddd30",
                  "shadowOpacity": 0.8,
                  "shadowRadius": 10,
                },
                false,
              ],
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#eee",
                  "fontFamily": "Montserrat_400Regular",
                  "fontSize": 20,
                },
                undefined,
              ]
            }
          >
            18 Fri
          </Text>
          <Text
            style={
              [
                {
                  "color": "#eee",
                  "fontFamily": "Montserrat_500Medium",
                  "fontSize": 16,
                },
                {
                  "marginTop": 10,
                },
              ]
            }
          >
            10.5
            <Text
              style={
                [
                  {
                    "color": "#eee",
                    "fontFamily": "Montserrat_200ExtraLight",
                    "fontSize": 14,
                  },
                  {
                    "marginHorizontal": 10,
                  },
                ]
              }
            >
              /
            </Text>
            16.8
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
